OpenRA.Game/Graphics/Animation.cs

```diff
@@ -199,5 +200,10 @@ public ISpriteSequence GetSequence(string sequenceName)
 		{
 			return sequenceProvider.GetSequence(name, sequenceName);
 		}
+
+		public string GetRandomExistingSequence(string[] sequences)
```

Comment:
This should take the random generator as an argument.  In its current state, somebody might accidentally use this for synced state (e.g. via PlayAnimationThen) and introduce a desync.